sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/secrets/Secret.java

```diff
@@ -0,0 +1,92 @@
+package com.mesosphere.sdk.dcos.secrets;
+
+import java.util.Collection;
+
+/**
+ * Secret class represents an HTTP API model for creating and updating a secret.
+ */
+public class Secret {
```

Comment:
How / where is this getting serialized out? You could use Jackson notations to simplify things if it's going to JSON.